modules/controlfreec/1.2/controlfreec.smk

```diff
@@ -0,0 +1,461 @@
+#!/usr/bin/env snakemake
+
+
+##### ATTRIBUTION #####
+
+
+# Original Author:  Jasper
+# Module Author:    Jasper
+# Contributors:     N/A
+
+
+##### SETUP #####
+
+
+# Import package with useful functions for developing analysis modules
+import oncopipe as op
+
+# Setup module and store module-specific configuration in `CFG`
+# `CFG` is a shortcut to `config["lcr-modules"]["controlfreec"]`
+CFG = op.setup_module(
+    name = "controlfreec",
+    version = "1.2",
+    subdirectories = ["inputs", "mpileup", "run", "outputs"]
+)
+
+
+
+# Define rules to be run locally when using a compute cluster
+localrules:
+    _controlfreec_input_bam,
+    _controlfreec_config,
+    _controlfreec_output,
+    _controlfreec_all
+
+
+##### RULES #####
+
+#### generate references ####
+# mappability tracks for hg19 and hg38 are available from the source
+rule _controlfreec_get_map_refs:
+    output:
+        tar = temp(CFG["dirs"]["inputs"] + "references/{genome_build}/freec/out100m2_{genome_build}.tar.gz"),
+        gem = CFG["dirs"]["inputs"] + "references/{genome_build}/freec/out100m2_{genome_build}.gem"
+    params:
+        provider = "ensembl",
+        url = lambda w: {"grch37": "http://xfer.curie.fr/get/7hZIk1C63h0/hg19_len100bp.tar.gz",
+                        "hg19": "http://xfer.curie.fr/get/7hZIk1C63h0/hg19_len100bp.tar.gz",
+                        "grch38": "http://xfer.curie.fr/get/vyIi4w8EONl/out100m2_hg38.zip",
+                        "hg38": "http://xfer.curie.fr/get/vyIi4w8EONl/out100m2_hg38.zip"}[w.genome_build],
+        command1 = lambda w: {"grch37": "tar -xvf ",
+                            "hg19": "tar -xvf ",
+                            "grch38": "unzip ",
+                            "hg38": "unzip "}[w.genome_build],
+        command2 = lambda w: {"grch37": " --wildcards --no-anchored 'out100m2*gem' && mv out100m2_hg19.gem ",
+                            "hg19": " --wildcards --no-anchored 'out100m2*gem' && mv out100m2_hg19.gem ",
+                            "grch38": " -d ",
+                            "hg38": " -d "}[w.genome_build],
+        command3 = lambda w: {"grch37": "out100m2_grch37.gem ",
+                            "hg19": "out100m2_hg19.gem ",
+                            "grch38": " ",
+                            "hg38": " "}[w.genome_build],
+        outdir = CFG["dirs"]["inputs"] + "references/{genome_build}/freec/"
+    wildcard_constraints:
+        genome_build = ".+(?<!masked)"
+    shell:
+        "wget -O {output.tar} {params.url} "
+        "&& {params.command1} {output.tar} {params.command2} {params.outdir}{params.command3}"
+
+# mappability tracks for hard-masked genomes need to be generated using GEM
+rule _download_GEM:
```

Comment:
It could be useful to make the location where GEM is installed user-configurable via the config. That way if they want to install software somewhere central that option is available.

Reply:
Okay I can add a SOFTWARE_DIR variable in the config